moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -859,6 +870,46 @@ bool TrajectoryExecutionManager::distributeTrajectory(const moveit_msgs::RobotTr
   return true;
 }
 
+bool TrajectoryExecutionManager::validate(const moveit_msgs::RobotTrajectory &trajectory) const
+{
+  ROS_DEBUG_NAMED("traj_execution", "validating trajectory with allowed_start_tolerance %g", allowed_start_tolerance_);
+
+  if (trajectory.joint_trajectory.points.empty())
+    return true;
+
+  robot_state::RobotStatePtr current_state;
+  if (!csm_->waitForCurrentState(1.0) || !(current_state = csm_->getCurrentState()))
+  {
+    ROS_WARN_NAMED("traj_execution", "Failed to validate trajectory: couldn't receive full current joint state within 1s");
+    return false;
+  }
+
+  const std::vector<double> &positions = trajectory.joint_trajectory.points.front().positions;
+  const std::vector<std::string> &joint_names = trajectory.joint_trajectory.joint_names;
+  const std::size_t n = joint_names.size();
+  if (positions.size() != n)
+    return false;
```

Comment:
Please add `ROS_WARN_NAMED` here.
This should never happen and we should not just silently fail.